### PR TITLE
[mdatagen] support using a different github project in mdatagen README issues list

### LINKED ISSUE #10484: [mdatagen] README template assumes components are in contrib repo
**Describe the bug**

When searching for open and closed issues against the component, the queries are done against `contrib`. `contrib` rarely (if ever) has labels for components in `core`, so the queries end up being invalid.

Using the [debug exporter](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/debugexporter#debug-exporter) as an example, the queries are as follows:

Open issues:
```
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Aexporter%2Fdebug
```
Closed issues:
```
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Aexporter%2Fdebug
```
There is no label `exporter/debug` in contrib, so there will never be any issues in either query.

### PR DESCRIPTION
**Pull request by atoulme**
#### Description
[mdatagen] support using a different github project in mdatagen README issues list

Uses a a new configuration property `github_project` on mdatagen configuration.

#### Link to tracking issue
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/9240

### CODE REVIEW

File: cmd/mdatagen/README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Detailed information about the schema of `metadata.yaml` can be found in [metada
 
 The `go:generate mdatagen` directive is usually defined in a `doc.go` file in the same package as the component, for example:
 ```go
-//go:generate mdatagen metadata.yaml
+//go:generate env GITHUB_PROJECT=open-telemetry/opentelemetry-collector mdatagen metadata.yaml
```
Comment (codeboten) [inline cmd/mdatagen/README.md:36]: i would prefer this be configurable in metadata.yaml, it avoids having configuration in two places

Comment (codeboten) [reply cmd/mdatagen/README.md:36]: See https://github.com/atoulme/opentelemetry-collector/pull/1

**Review by codeboten — APPROVED** [review]

File: cmd/mdatagen/README.md (hunk 1 above)
Comment (dmitryax) [inline cmd/mdatagen/README.md:36]: Can we remove this if it's configured in metadata.yaml now?

**Comment by dmitryax** [comment]
ALso please update the description

File: cmd/mdatagen/README.md (hunk 1 above)
Comment (atoulme) [reply cmd/mdatagen/README.md:36]: ah missed a spot sorry
```suggestion
//go:generate mdatagen metadata.yaml
```

**Pull request merged**